gems/aws-sdk-s3/lib/aws-sdk-s3/object_multipart_copier.rb

```diff
@@ -20,13 +20,13 @@ class ObjectMultipartCopier
       #   will be constructed from the given `options' hash.
       # @option [Integer] :thread_count (10) Number of concurrent threads to
       #   use for copying parts.
+      # @option [Boolean] :use_source_parts (false) Use part sizes defined on
```

Comment:
Whats the reason for adding this option?

Reply:
If copying or moving an object that is already multipart, this does not re-part the object, instead re-using the part definitions on the original. That means the etag and any checksums will not change.

Reply:
It's possible however that some may be using this to intentionally re-part an object that has part sizes that are not convenient, so I thought it made sense to make that behavior optional

Reply:
this is especially useful if the source object has parts with varied sizes, like the output from `EC2::Client#create_store_image_task`

Reply:
I think this is important context to put in the documentation for the option.

Reply:
I'll add a summary, sure.